optuna/study.py

```diff
@@ -677,6 +677,27 @@ def load_study(
     return Study(study_name=study_name, storage=storage, sampler=sampler, pruner=pruner)
 
 
+def delete_study(
+        study_name,  # type: str
+        storage=None,  # type: Union[None, str, storages.BaseStorage]
```

Comment:
IMO, `storage=None` is meaningless for this method (it'll create an empty study and delete it immediately).
How about removing the default value of `storage` and modifying the type hint to `Union[str, storages.BaseStorage]`? > @c-bata @toshihikoyanase @g-votte

Reply:
Agree. Does it apply to `create_study` and `load_study`?

Reply:
I agree with @sile.
I think it is similar to `storage` argument of [`load_study`](https://optuna.readthedocs.io/en/stable/reference/study.html#optuna.study.load_study).

Reply:
I think that the default storage argument of `create_study` (i.e., `create_study(storage=None)`) is useful to create an empty in-memory storage.

Reply:
Thank you. I applied your feedbacks and pushed it.

Reply:
Hmm, `storage` still has the default value: https://github.com/pfnet/optuna/pull/558/files#diff-4e020643693eb296936f5dd7194bd6d6R682

Reply:
Oh, I missed!